site/zenodo_rdm/assets/semantic-ui/js/zenodo_rdm/src/citations/sanitizer.js

```diff
@@ -0,0 +1,151 @@
+// This file is part of Zenodo.
+// Copyright (C) 2022 CERN.
+//
+// Zenodo is free software; you can redistribute it
+// and/or modify it under the terms of the GNU General Public License as
+// published by the Free Software Foundation; either version 2 of the
+// License, or (at your option) any later version.
+//
+// Zenodo is distributed in the hope that it will be
+// useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Zenodo; if not, write to the
+// Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+// MA 02111-1307, USA.
+//
+// In applying this license, CERN does not
+// waive the privileges and immunities granted to it by virtue of its status
+// as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+function providerNamesFilter(relationship) {
+  var providerNames = [];
+  relationship.metadata.History.forEach(function (linkHistory) {
+    linkHistory.LinkProvider.forEach(function (provider) {
+      if (!providerNames.includes(provider.Name)) {
+        providerNames.push(provider.Name);
+      }
+    });
+  });
+  return providerNames.join(", ");
+}
+
+function creatorNamesFilter(relationship) {
+  var creatorNames = "";
+  var creators = relationship.metadata.Source.Creator;
+  if (creators) {
+    creatorNames = creators[0].Name;
+    if (creators.length == 2) {
+      creatorNames = creatorNames + " & " + creators[1].Name;
+    } else if (creators.length > 2) {
+      creatorNames = creatorNames + " et al.";
+    }
+  }
+  return creatorNames;
+}
+
+function doiUrlFilter(relationship) {
+  var doiUrl = "";
+  var url = "";
```

Comment:
```suggestion
  const doiUrl = undefined;
  const url = undefined;
```

Reply:
These can't be `const` as they are reassigned later in the loop.

Reply:
Maybe use `let` instead :)